test/CMakeLists.txt

```diff
@@ -57,15 +57,30 @@ add_custom_target(run-onedpl-tests
     DEPENDS build-onedpl-tests
     COMMENT "Build and run all oneDPL tests")
 
-macro(onedpl_add_test test_source_file switch_off_checked_iterators)
+macro(onedpl_add_test test_source_file switch_off_checked_iterators is_std_test)
 
     set(_build_type_in_lower "unknown")
     if (CMAKE_BUILD_TYPE)
         string(TOLOWER "${CMAKE_BUILD_TYPE}" _build_type_in_lower)
     endif()
 
-    get_filename_component(_test_name ${test_source_file} NAME)
-    string(REPLACE "\.cpp" "" _test_name ${_test_name})
+    if (${is_std_test})
+        #process std tests: target name equal to parent folders names plus cpp file name without extension
+        file(RELATIVE_PATH _test_name "${CMAKE_CURRENT_SOURCE_DIR}" "${test_source_file}")
+        string(REPLACE "\.cpp" "" _test_name ${_test_name})
+        string(REPLACE "/" ";" _test_name_list ${_test_name})
```

Comment:
To make sure that the received path uses "/" as a separator on both Linux and Windows, you can use `TO_CMAKE_PATH` option while receiving the path through `file` command: https://cmake.org/cmake/help/v3.15/command/file.html#path-conversion.

Reply:
This change has been rolled back.